docs/install/Knative-with-Minikube.md

```diff
@@ -89,58 +89,72 @@ rerun the command to see the current status.
-> command to view the component's status updates in real time. Use CTRL+C to
-> exit watch mode.
-
-Now you can deploy an app to your newly created Knative cluster.
+## Installing Knative
+
+The following commands install all available Knative components as well as the
+standard set of observability plugins. To customize your Knative installation,
+see [Performing a Custom Knative Installation](./Knative-custom-install.md).
+
+1. If you are upgrading from Knative 0.3.x: Update your domain and static IP
+   address to be associated with the LoadBalancer `istio-ingressgateway` instead
+   of `knative-ingressgateway`. Then run the following to clean up leftover
+   resources:
+
+   ```
+   kubectl delete svc knative-ingressgateway -n istio-system
+   kubectl delete deploy knative-ingressgateway -n istio-system
+   ```
+
+   If you have the Knative Eventing Sources component installed, you will also
+   need to delete the following resource before upgrading:
+
+   ```
+   kubectl delete statefulset/controller-manager -n knative-sources
+   ```
+
+   While the deletion of this resource during the upgrade process will not
+   prevent modifications to Eventing Source resources, those changes will not be
+   completed until the upgrade process finishes.
+
+1. Run the `kubectl apply` command once with the `-l knative.dev/crd-install=true`
+   flag to install the CRDs first. (This prevents race conditions, which
+   cause intermittent install errors.)
+
+   ```bash
+   kubectl apply -l knative.dev/crd-install=true \
+   --filename https://github.com/knative/serving/releases/download/v0.5.0/serving.yaml \
+   --filename https://github.com/knative/build/releases/download/v0.5.0/build.yaml \
+   --filename https://github.com/knative/eventing/releases/download/v0.5.0/release.yaml \
+   --filename https://github.com/knative/eventing-sources/releases/download/v0.5.0/eventing-sources.yaml \
+   --filename https://github.com/knative/serving/releases/download/v0.5.0/monitoring.yaml \
+   --filename https://raw.githubusercontent.com/knative/serving/v0.5.0/third_party/config/build/clusterrole.yaml
+   ```
+
+1. Then run the `kubectl apply` command without the `-l` flag to complete the
+   install of Knative and its dependencies:
+
+   ```bash
+   kubectl apply --filename https://github.com/knative/serving/releases/download/v0.5.0/serving.yaml \
+   --filename https://github.com/knative/build/releases/download/v0.5.0/build.yaml \
+   --filename https://github.com/knative/eventing/releases/download/v0.5.0/release.yaml \
+   --filename https://github.com/knative/eventing-sources/releases/download/v0.5.0/eventing-sources.yaml \
+   --filename https://github.com/knative/serving/releases/download/v0.5.0/monitoring.yaml \
+   --filename https://raw.githubusercontent.com/knative/serving/v0.5.0/third_party/config/build/clusterrole.yaml
+   ```
+
+   > **Note**: For the v0.4.0 release and newer, the `clusterrole.yaml` file is
```

Comment:
Well, since you install clusterroles to make serving and build interact, you don't need it if you don't plan to use Serving and Build together.

This also falls into the category of "should probably be addressed in a separate PR if we want to make updates to that."

Reply:
sounds good. just concerned with how it is today vs "how we want it and when that happens"